config/blockchain_network_config.go

```diff
@@ -0,0 +1,140 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+)
+
+type NetworkSelected struct {
+	NetworkName             string
+	EthereumJSONRPCEndpoint string
+	NetworkId               string
+	RegistryAddressKey      string
+}
+
+const (
+	DefaultBlockChainNetworkConfig string = `
+{
+  "local":{
+    "ethereum_json_rpc_endpoint":"http://localhost:8545",
+    "network_id":"999999",
+    "registry_address_key":"0x4e74fefa82e83e0964f0d9f53c68e03f7298a8b2"
+  },
+  "kovan":{
+    "ethereum_json_rpc_endpoint":"https://kovan.infura.io",
+    "network_id":"42"
+  },
+  "ropsten":{
+    "ethereum_json_rpc_endpoint":"https://ropsten.infura.io",
+    "network_id":"3"
+  },
+  "rinkeby":{
+    "ethereum_json_rpc_endpoint":"https://rinkeby.infura.io",
+    "network_id":"4"
+  },
+  "main":{
+    "ethereum_json_rpc_endpoint":"https://mainnet.infura.io",
+    "network_id":"1"
+  }
+}`
+	BlockChainNetworkFileName  = "blockchain_network_config.json"
+	EthereumJsonRpcEndpointKey = "ethereum_json_rpc_endpoint"
+	NetworkId                  = "network_id"
+	RegistryAddressKey         = "registry_address_key"
+	//Read this from a config file path todo
+	RegistryJsonFileName = "resources/blockchain/node_modules/singularitynet-platform-contracts/networks/Registry.json"
+)
+
+var networkSelected *NetworkSelected
+
+func determineNetworkSelected(data []byte) {
+	dynamicBinding := map[string]interface{}{}
+	networkName := GetString(BlockChainNetworkSelected)
+	err := json.Unmarshal(data, &dynamicBinding)
+	if err != nil {
+		panic(fmt.Sprintf("cannot parse the JSON configuation file %v for the network %v  to determine the network selected: %v",
```

Comment:
I would replace `panic` calls here and below by returning `err` to the caller and further and at the end return it from `setAndValidateBlockChainNetworkDetails` and from `Validate` function in `config.go`. Thus all validation errors will be handled by similar way but I don't insist on it.

Reply:
Will do